R/object_name_linter.R

```diff
@@ -44,6 +44,11 @@ object_name_xpath <- local({
 #' @param styles A subset of
 #'   \Sexpr[stage=render, results=rd]{lintr:::regexes_rd}. A name should
 #'   match at least one of these styles.
+#' @param regexes A (possibly named) character vector specifying a custom naming convention.
+#'   If named, the names will be used in the lint message. Otherwise, "custom" will be used as a name for the style.
+#'   Quotes (`` `"' ``) and specials (`%` and trailing `<-`) are not considered part of the name to be matched.
```

Comment:
should this (Quotes...matched) be mentioned in the above details? it may fit more with the discussion of S3 patterns, WDYT?

Reply:
Hmm sounds good. Also, the docs regading the auto-naming are out-of date 🤦🏻‍♂️